src/data/nat/prime/min_fac.lean

```diff
@@ -0,0 +1,35 @@
+/-
+Copyright (c) 2019 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import data.nat.prime
+import tactic.fin_cases
+
+/-!
+# Lemmas about min_fac
+
+These lemmas don't work in `data/nat/prime.lean`, because they use `fin_cases` in the proofs,
+which would introduce a cyclic import.
+-/
+
+namespace nat
+
+lemma min_fac_le_div {n : ℕ} (pos : 0 < n) (np : ¬ prime n) : min_fac n ≤ n / min_fac n :=
+begin
+  by_cases h : n = 1,
+  { subst h, simp, },
+  { replace h : 2 ≤ n := le_of_not_gt (λ hgt, h $ le_antisymm (nat.le_pred_of_lt hgt) (nat.succ_le_of_lt pos)),
+    have d : min_fac n ∣ n := min_fac_dvd n,
+    apply min_fac_le_of_dvd,
+    { by_contradiction,
+      rw [not_le] at a,
+      -- replace with `nat_cases` when it arrives
+      replace a : n / min_fac n ∈ [0,1] := list.Ico.mem.mpr ⟨zero_le _, a⟩,
+      fin_cases a,
+      { exact ne_of_gt pos (eq_zero_of_div_eq_zero a d (min_fac_pos _)) },
+      { exact np (nat.prime_def_min_fac.2 ⟨h, (eq_of_dvd_quot_one d a)⟩) } },
+    exact div_dvd_of_dvd d, }
+end
+
```

Comment:
Here's a marginally shorter proof that does not use `fin_cases`
```lean
lemma min_fac_le_div {n : ℕ} (pos : 0 < n) (np : ¬ prime n) : min_fac n ≤ n / min_fac n := 
match min_fac_dvd n with
| ⟨0, h0⟩     := absurd pos $ by rw [h0, mul_zero]; exact dec_trivial
| ⟨1, h1⟩     := begin
  rw mul_one at h1,
  rw [prime_def_min_fac, not_and_distrib, ← h1, eq_self_iff_true, not_true, or_false, not_le] at np,
  rw [le_antisymm (le_of_lt_succ np) (succ_le_of_lt pos), min_fac_one, nat.div_one]
end
| ⟨(x+2), hx⟩ := begin
  conv_rhs { congr, rw hx },
  rw [nat.mul_div_cancel_left _ (min_fac_pos _)],
  exact min_fac_le_of_dvd dec_trivial ⟨min_fac n, by rwa mul_comm⟩
end

```

Reply:
I've gone with this one. I like the pattern match, which makes the proof clearer.

Reply:
I'm always confused about those `begin` at end of line. I don't care too much about style, but I thought the convention was to put it at the beginning of a new line...

Reply:
I made this stylistic change (move the `begin`s to new lines) when I copied into the PR.